File: JS/day 11/secondLarge.js
//question 1:
//find second largest element from an array
function findSecondLargest(arr) {
    if (arr.length < 2) {
        return null; // Not enough elements
    }

    let largest = arr[0];
    let secondLargest;

    for (let i = 1; i < arr.length; i++) {
        if (arr[i] > largest) {
            secondLargest = largest;
            largest = arr[i];
        } else if (arr[i] > secondLargest && arr[i] !== largest) {
            secondLargest = arr[i];
        }
    }

    return secondLargest;
}

// Example usage:
const array = [10, 5, 8, 12, 15, 7];
console.log(findSecondLargest(array));  // Output: 12

File: DSA/JAVA/day 04/SecondLarge.java
/*
given an array find the second largest number
*/
class SLarge{
    public static int SecondLargestNumber(int[] arr) {

        int largest = arr[0];
        int secondLargest = arr[1];

        for (int i = 0; i < arr.length; i++) {
            if (arr[i] > largest) {
                secondLargest = largest;
                largest = arr[i];
            } else if (arr[i] < largest && arr[i] > secondLargest) {
                secondLargest = arr[i];
            }
        }
        return secondLargest;
    }
}

public class SecondLarge {
    public static void main(String[] args) {
        int[] arr = { 1, 2, 3, 4, 5, 6, 7, 8, 9, 10 };
        System.out.println(SLarge.SecondLargestNumber(arr));
    }
}
